edge/pkg/clusterd/edgecluster_status_reporter.go

```diff
@@ -0,0 +1,217 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clusterd
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	utilwait "k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog/v2"
+
+	"github.com/kubeedge/beehive/pkg/core"
+	beehiveContext "github.com/kubeedge/beehive/pkg/core/context"
+	"github.com/kubeedge/beehive/pkg/core/model"
+	edgeclustersv1 "github.com/kubeedge/kubeedge/cloud/pkg/apis/edgeclusters/v1"
+	edgeapi "github.com/kubeedge/kubeedge/common/types"
+	"github.com/kubeedge/kubeedge/edge/pkg/clusterd/config"
+	"github.com/kubeedge/kubeedge/edge/pkg/clusterd/helper"
+	"github.com/kubeedge/kubeedge/edge/pkg/clusterd/util"
+	"github.com/kubeedge/kubeedge/edge/pkg/common/message"
+	"github.com/kubeedge/kubeedge/edge/pkg/common/modules"
+	"github.com/kubeedge/kubeedge/edge/pkg/edgehub"
+)
+
+const (
+	MissionCrdFile     = "mission_v1.yaml"
+	EdgeClusterCrdFile = "edgecluster_v1.yaml"
+)
+
+var initEdgeCluster edgeclustersv1.EdgeCluster
+
+type EdgeClusterStatusReporter struct {
+	clusterd                        *clusterd
+	edgeClusterStatusUpdateInterval time.Duration
+	missionDeployer                 *MissionDeployer
+	registrationCompleted           bool
+}
+
+func NewEdgeClusterStatusReporter(c *clusterd, md *MissionDeployer) *EdgeClusterStatusReporter {
+	return &EdgeClusterStatusReporter{
+		clusterd:                        c,
+		missionDeployer:                 md,
+		edgeClusterStatusUpdateInterval: time.Duration(config.Config.EdgeClusterStatusUpdateInterval) * time.Second,
+		registrationCompleted:           false,
+	}
+}
+
+func (esr *EdgeClusterStatusReporter) initialEdgeCluster() (*edgeclustersv1.EdgeCluster, error) {
+	var ec = &edgeclustersv1.EdgeCluster{}
+
+	if err := esr.prepareCluster(); err != nil {
+		return nil, err
+	}
+
+	ec.Name = config.Config.Name
+	ec.Spec.Kubeconfig = config.Config.Kubeconfig
+	ec.Spec.KubeDistro = config.Config.KubeDistro
+
+	ec.Labels = map[string]string{
+		// Kubernetes built-in labels
+		v1.LabelHostname: ec.Name,
+
+		// KubeEdge specific labels
```

Comment:
KubeEdge --> Fornax

Reply:
Actually, they are Kubeedge specific. These labels are used in edge nodes as well.